### DISCUSSION TITLE
Alpha6: javax.inject.@Inject present in quarkus-junit5 dependency

### DISCUSSION THREAD

**Discussion by luca-bassoricci** [quarkusio/quarkus#31890] [Community]
I was upgrading to Alpha6 and some tests are failing due to wrong import of `javax.inject.Inject`.
I found `javax.inject.@Inject` annotation is still present in `quarkus-junit5` extension: is that a mistake and `javax.inject:javax.inject` shall be excluded by `quarkus-junit5` module?

![javax-inject-quarkus-junit5](https://user-images.githubusercontent.com/73338789/225564620-a6b800fd-c03c-4d8e-a820-96f4cc25a2ed.png)

**Comment by gsmet** [comment]
Yes, that's unfortunate but, for now, having the `@javax.inject.Inject` annotation around is required by our Maven resolver (which we need when executing tests).

We have https://github.com/quarkusio/quarkus/issues/27593 to track it but I don't think it will be solved for the final. I would recommend to make sure the jakarta version arrives before the javax one in your IDE.

@manovotn @Ladicek didn't we add something in ArC to warn users about this?

I'm closing as tracked in another issue.

**Reply by gsmet** [reply]
Ah I thought it was an issue but it's a discussion so keeping it open :).

**Reply by Ladicek** [reply]
Yeah, the ArC extension should validate known-wrong `@Inject` annotations, `javax.inject.Inject` being one of them. Not sure if that happens here or not. It seems possible to me that annotations in the test class are not validated, as the test class likely isn't in the application index, but not sure.

**Reply by luca-bassoricci** [reply]
@Ladicek invalid `@Inject` are correctly reported also during tests
